rootfs/etc/nginx/lua/monitor.lua

```diff
@@ -66,12 +68,13 @@ function _M.init_worker()
 end
 
 function _M.call()
-  if #metrics_batch >= MAX_BATCH_SIZE then
+  local metrics_size = #metrics_batch
+  if metrics_size >= MAX_BATCH_SIZE then
     ngx.log(ngx.WARN, "omitting metrics for the request, current batch is full")
     return
   end
 
-  table.insert(metrics_batch, metrics())
+  metrics_batch[metrics_size + 1] = metrics()
```

Comment:
Does this custom table implementation not have `insert`?

Reply:
it has the `insert` method, but the `table.insert` is an O(n) operation due to the lj_tab_len call inside LuaJIT.
We can avoid it in hot loops for better performance.